File: sdks/php/src/Models/PdfSplitResult.php
<?php

declare(strict_types=1);

namespace Renamed\Models;

/**
 * Result of PDF split operation.
 */
final class PdfSplitResult
{
    /**
     * @param SplitDocument[] $documents Split documents.
     */
    public function __construct(
        /** Original filename. */
        public readonly string $originalFilename,
        /** Split documents. */
        public readonly array $documents,
        /** Total number of pages in original document. */
        public readonly int $totalPages,
    ) {}

    /**
     * Create instance from API response array.
     *
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        $documents = array_map(
            fn(array $doc) => SplitDocument::fromArray($doc),
            $data['documents'] ?? []
        );

        return new self(
            originalFilename: $data['originalFilename'],
            documents: $documents,
            totalPages: (int) $data['totalPages'],
        );
    }

    /**
     * Convert to array.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'originalFilename' => $this->originalFilename,
            'documents' => array_map(fn(SplitDocument $doc) => $doc->toArray(), $this->documents),
            'totalPages' => $this->totalPages,
        ];
    }
}

File: sdks/csharp/Renamed/Models/PdfSplitResult.cs
using System.Text.Json.Serialization;

namespace Renamed.Sdk.Models;

/// <summary>
/// A single document resulting from PDF split operation.
/// </summary>
public sealed class SplitDocument
{
    /// <summary>
    /// Document index (0-based).
    /// </summary>
    [JsonPropertyName("index")]
    public int Index { get; init; }

    /// <summary>
    /// Suggested filename for this document.
    /// </summary>
    [JsonPropertyName("filename")]
    public required string Filename { get; init; }

    /// <summary>
    /// Page range included in this document (e.g., "1-5").
    /// </summary>
    [JsonPropertyName("pages")]
    public required string Pages { get; init; }

    /// <summary>
    /// URL to download this document.
    /// </summary>
    [JsonPropertyName("downloadUrl")]
    public required string DownloadUrl { get; init; }

    /// <summary>
    /// Size of the document in bytes.
    /// </summary>
    [JsonPropertyName("size")]
    public long Size { get; init; }
}

/// <summary>
/// Result of PDF split operation.
/// </summary>
public sealed class PdfSplitResult
{
    /// <summary>
    /// Original filename of the uploaded PDF.
    /// </summary>
    [JsonPropertyName("originalFilename")]
    public required string OriginalFilename { get; init; }

    /// <summary>
    /// Collection of split documents.
    /// </summary>
    [JsonPropertyName("documents")]
    public required IReadOnlyList<SplitDocument> Documents { get; init; }

    /// <summary>
    /// Total number of pages in the original document.
    /// </summary>
    [JsonPropertyName("totalPages")]
    public int TotalPages { get; init; }
}
